pilot/docker/Dockerfile.proxyv2

```diff
@@ -1,9 +1,98 @@
 ARG BASE_DISTRIBUTION=debug
-
 # Version is the base image version from the TLD Makefile
 ARG BASE_VERSION=latest
 
+FROM ubuntu:focal as builder
+
+ARG SDK_VERSION="2.17.100.3"
+ARG SGX_SDK_INSTALLER=sgx_linux_x64_sdk_${SDK_VERSION}.bin
+ARG DCAP_VERSION="1.14.100.3"
+
+RUN apt-get update && \
+    env DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    wget \
+    unzip \
+    protobuf-compiler \
+    libprotobuf-dev \
+    build-essential \
+    cmake \
+    pkg-config \
+    gdb \
+    vim \
+    python3 \
+    git \
+    gnupg \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Add 01.org to apt for SGX packages
+# hadolint ignore=DL4006
+RUN echo "deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main" >> /etc/apt/sources.list.d/intel-sgx.list \
+  && wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - \
+# Install SGX PSW
+  && apt-get update \
+  && apt-get install --no-install-recommends -y \
+    libsgx-enclave-common=${SDK_VERSION}-focal1 \
+    libsgx-launch=${SDK_VERSION}-focal1 \
+    libsgx-launch-dev=${SDK_VERSION}-focal1 \
+    libsgx-epid=${SDK_VERSION}-focal1 \
+    libsgx-epid-dev=${SDK_VERSION}-focal1 \
+    libsgx-quote-ex=${SDK_VERSION}-focal1 \
+    libsgx-quote-ex-dev=${SDK_VERSION}-focal1 \
+    libsgx-urts=${SDK_VERSION}-focal1 \
+    libsgx-uae-service=${SDK_VERSION}-focal1 \
+    libsgx-ae-epid=${SDK_VERSION}-focal1 \
+    libsgx-ae-le=${SDK_VERSION}-focal1 \
+    libsgx-ae-pce=${SDK_VERSION}-focal1 \
+    libsgx-ae-qe3=${DCAP_VERSION}-focal1 \
+    libsgx-ae-qve=${DCAP_VERSION}-focal1 \
+    libsgx-dcap-ql=${DCAP_VERSION}-focal1 \
+    libsgx-dcap-ql-dev=${DCAP_VERSION}-focal1 \
+    libsgx-dcap-default-qpl-dev=${DCAP_VERSION}-focal1 \
+    libsgx-pce-logic=${DCAP_VERSION}-focal1 \
+    libsgx-qe3-logic=${DCAP_VERSION}-focal1 \
+    libsgx-dcap-default-qpl=${DCAP_VERSION}-focal1 \
+    libengine-pkcs11-openssl
+
+# SGX SDK is installed in /opt/intel directory.
+WORKDIR /opt/intel
+
+# Install SGX SDK
+# hadolint ignore=DL4006
+RUN wget https://download.01.org/intel-sgx/sgx-linux/2.15/distro/ubuntu20.04-server/$SGX_SDK_INSTALLER \
+  && chmod +x  $SGX_SDK_INSTALLER \
+  && echo "yes" | ./$SGX_SDK_INSTALLER \
+  && rm $SGX_SDK_INSTALLER \
+  && ls -l /opt/intel/
+
+# Intel crypto-api-toolkit prerequisites
+#https://github.com/intel/crypto-api-toolkit#software-requirements
+RUN set -x && apt-get update \
+  && apt-get install --no-install-recommends -y \
+    dkms libprotobuf17 autoconf \
+    autotools-dev libc6-dev \
+    libtool build-essential \
+    opensc sudo \
+    automake \
+  && apt-get clean \
+  && git clone https://github.com/intel/crypto-api-toolkit.git \
+  && cd /opt/intel/crypto-api-toolkit \
+  # disable building tests
+  && sed -i -e 's;test;;g' ./src/Makefile.am \
+  && ./autogen.sh \
+  && ./configure --enable-dcap  \
```

Comment:
Are you signing the enclave with the CTK test key? You need to sign using a real private key if someone needs to attest the quote (since the quote contains the MRSIGNER value). Is that the case since you build with `--enable-dcap`?

Reply:
> Are you signing the enclave with the CTK test key? You need to sign using a real private key if someone needs to attest the quote (since the quote contains the MRSIGNER value). Is that the case since you build with `--enable-dcap`?

We are doing this temporarily. We will add a flag such as `--ctk-privkey="private key path"` in the build phase later, for users to choose to use their own key or the default key of CTK